runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/DefaultJobBundleFactory.java

```diff
@@ -342,22 +343,14 @@ public RemoteBundle getBundle(
       // TODO: Consider having BundleProcessor#newBundle take in an OutputReceiverFactory rather
       // than constructing the receiver map here. Every bundle factory will need this.
 
-      if (environmentExpirationMillis == 0 && !loadBalanceBundles) {
-        return currentClient.processor.newBundle(
-            getOutputReceivers(currentClient.processBundleDescriptor, outputReceiverFactory)
-                .build(),
-            stateRequestHandler,
-            progressHandler);
-      }
```

Comment:
I discovered another issue in this code path. `ref()` is not called on the client in this case. Also, `unref()` will never be called because the code below does not insert the call to the bundle close. `close()` is only ever called on `unref()`.

Reply:
This is the original code path that was used when no reference counting is required. What was the problem with it?

Reply:
I suppose it was working correctly due to the initial ref() when creating the environment. Still, it doesn't hurt to remove this bit because it just adds a specialized execution path to the generalized version. The ref business is already complex enough. If you think differently feel free to add it back, but IMHO this is easier to understand.

Reply:
During the initial review of the bundle load balancing the idea came up to have two implementations, one with refing, one without. I think that would be the best improvement moving forward.